packages/headless/src/utils/vitest.ts

```diff
@@ -0,0 +1,36 @@
+import { vi, type Awaitable } from "vitest";
+
+type Callback = () => Awaitable<void>;
+
+/**
+ * Mocks the following vue lifecycle functions:
+ * - onBeforeMount
+ * - onMounted
+ * - onBeforeUnmount
+ * - onUnmounted
+ *
+ * `onBeforeMount` and `onMounted` callbacks are executed immediately.
+ * `onBeforeUnmount` and `onUnmounted` are executed when the returned callback is run.
+ * @returns a callback to trigger the run of `onBeforeUnmount` and `onUnmounted`
+ */
+export const mockVueLifecycle = () => {
+  const { callbacks } = vi.hoisted(() => ({
+    callbacks: {
+      onBeforeUnmountedCb: null as Callback | null,
+      onUnmountedCb: null as Callback | null,
+    },
+  }));
+
+  vi.mock("vue", async (original) => ({
+    ...((await original()) as object),
+    onBeforeMount: (cb: Callback) => cb(),
```

Comment:
```suggestion
    onBeforeMount: vi.fn().mockImplementation((cb: Callback) => cb())
```

Should we use vi.fn() here so they can be spyed on?